core/src/test/java/org/apache/accumulo/core/file/blockfile/cache/TestCachedBlockQueue.java

```diff
@@ -18,14 +18,65 @@
 package org.apache.accumulo.core.file.blockfile.cache;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
-import java.util.LinkedList;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicLong;
 
 import org.apache.accumulo.core.file.blockfile.cache.lru.CachedBlockQueue;
 import org.junit.Test;
 
 public class TestCachedBlockQueue {
 
+  @Test
+  public void testLargeBlock() {
+    CachedBlockQueue queue = new CachedBlockQueue(10000L, 1000L);
+    CachedBlock cb1 = new CachedBlock(10001L, "cb1", 1L);
+    cb1.recordSize(new AtomicLong(10001L));
+    assertTrue(queue.add(cb1));
+  }
+
+  @Test
+  public void testAddNewerBlock() {
+    CachedBlockQueue queue = new CachedBlockQueue(10000L, 1000L);
+
+    CachedBlock cb1 = new CachedBlock(5000L, "cb1", 1L);
+    cb1.recordSize(new AtomicLong(5000L));
+
+    CachedBlock cb2 = new CachedBlock(5000, "cb2", 2L);
+    cb2.recordSize(new AtomicLong(5000L));
+
+    CachedBlock cb3 = new CachedBlock(5000, "cb3", 3L);
+    cb3.recordSize(new AtomicLong(5000L));
```

Comment:
The `recordSize()` calls could be removed and called in the constructor of the test class.